### DISCUSSION TITLE
Error: "Attempting to resetState while current state is destroyed" (5.6.0)

### DISCUSSION THREAD

**Discussion by moxie-coder** [HaxeFlixel/flixel#3031] [Q&A] [answered]
I would get a “Attempting to resetState while current state is destroyed” error from legacy code, I don’t really get how it works now with the latest flixel

**Comment by Geokureli** [comment, marked as answer]
can you show how you're switching states?

@CaptainBaldi had the [same issue, in the discord](https://discord.com/channels/162395145352904705/165234904815239168/1203369545834692729), maybe there's a similar solution

**Reply by moxie-coder** [reply]
same sort of thing he was doing, I’ll try the solution you mentioned in the discord

**Reply by Geokureli** [reply]
for posterity can you share the relevant code, anyway?

**Reply by moxie-coder** [reply]
forgot to send it, here
```haxe
	public static function switchState(nextState:FlxState) {
		// Custom made Trans in
		var curState:Dynamic = FlxG.state;
		var leState:MusicBeatState = curState;
		if(!FlxTransitionableState.skipNextTransIn) {
			leState.openSubState(new CustomFadeTransition(0.6, false));
			if(nextState == FlxG.state) {
				CustomFadeTransition.finishCallback = function() {
					FlxG.resetState();
				};
				//trace('resetted');
			} else {
				CustomFadeTransition.finishCallback = function() {
					FlxG.switchState(nextState);
				};
				//trace('changed state');
			}
			return;
		}
		FlxTransitionableState.skipNextTransIn = false;
		FlxG.switchState(nextState);
	}

	public static function resetState() {
		MusicBeatState.switchState(FlxG.state);
	}
```

**Reply by Geokureli** [reply]
for anyone else doing stuff like this, the recommend fix was to override `startOutro` in your state for handling outro transitions rather than doing something like this

**Reply by Geokureli** [reply]
related: https://github.com/ShadowMario/FNF-PsychEngine/pull/14126
